lib/matplotlib/backends/backend_tkagg.py

```diff
@@ -529,21 +532,44 @@ def __init__(self, canvas, num, window):
         self.window.withdraw()
         self.set_window_title("Figure %d" % num)
         self.canvas = canvas
-        self._num =  num
-        if matplotlib.rcParams['toolbar']=='toolbar2':
-            self.toolbar = NavigationToolbar2TkAgg( canvas, self.window )
-        else:
-            self.toolbar = None
-        if self.toolbar is not None:
-            self.toolbar.update()
         self.canvas._tkcanvas.pack(side=Tk.TOP, fill=Tk.BOTH, expand=1)
+        self._num = num
+
+        self.toolmanager = self._get_toolmanager()
+        self.toolbar = self._get_toolbar()
+        self.statusbar = None
+
+        if matplotlib.rcParams['toolbar'] == 'toolmanager':
+            tools.add_tools_2_toolmanager(self.toolmanager)
+            tools.add_tools_2_container(self.toolbar)
```

Comment:
ah, just now noticed that "tools" is a rename of "backend_tools" module. What's the reason for the rename? The name "tools" doesn't click for me as a module, but as an object, so I saw that as a method call.